portal/src/main/webapp/WEB-INF/web.xml

```diff
@@ -647,7 +644,20 @@
         <url-pattern>/ln</url-pattern>
     </servlet-mapping>
 
-
+    <servlet>
+        <servlet-name>SinglePageApplication</servlet-name>
+        <jsp-file>/index.jsp</jsp-file>
+    </servlet>
+    <!--mapping-->
+    <servlet-mapping>
```

Comment:
@inodb note that every time we add a new route to SPA frontend, we have to make sure that tompcat with resolve that route to the SPA page (index.jsp).   This is kind of cumbersome and can be dealt with more generally but would require some learning/tinkering.  The strategy I've seen is to get server to resolve 404s to the SAP.  Then frontend can either match the route or show a 404 if there is no match.  But we DON'T want to resolve api 404s to SPA.   Food for thought.

Reply:
resolved ☝️